lib/graphql/batch/execution_strategy.rb

```diff
@@ -1,66 +0,0 @@
-module GraphQL::Batch
-  class ExecutionStrategy < GraphQL::Query::SerialExecution
-    attr_accessor :disable_batching
-
-    def execute(_, _, query)
-      as_promise(super).sync
-    rescue GraphQL::InvalidNullError => err
-      err.parent_error? || query.context.errors.push(err)
-      nil
-    ensure
-      GraphQL::Batch::Executor.current.clear
```

Comment:
Oh yikes, that's bad! Would any of these approaches work to save users from thinking about it? 

- Clear the executor _before_ starting a query 
- Provide a `begin ... ensure` wrapper for user code like `GraphQL::Batch.isolate { ... }` (but with a human-friendly name) 
- `instrument(:query, GraphQL::Batch::QueryInstrumentation)` to ensure cleanup (this doesn't work yet -- I bet the after-query hook isn't called if there's an error, but that could be fixed)